configure.ac

```diff
@@ -129,7 +129,7 @@ elif test "x$have_ubuntuindicator" != "xyes" &&
 
 else
 
-    AC_MSG_ERROR([Either Ayatana Indicators or Ubuntu Indicators are required to build mate-indicator-applet.])
+    AC_MSG_ERROR([Either Ayatana Indicators (--with-ayatana-indicators) or Ubuntu Indicators (--with-ubuntu-indicators) are required to build mate-indicator-applet.])
```

Comment:
This doesn't look right to me.  IIUC, the problem is that the case where you have *both* Ayatana and Ubuntu indicators is not handled, so results in an error telling you you don't have any instead of both.

IMO the whole test chain should be modified to be simpler and test it all, something like that:
```shell
if test "x$with_ayatana_indicators" = "xyes"; then
    use_ayatanaindicator=yes;
    indicator_enforced=yes;
elif test "x$with_ubuntu_indicators" = "xyes"; then
    use_ubuntuindicator=yes;
    indicator_enforced=yes;
elif test "x$have_ayatanaindicator" = "xyes"; then
    use_ayatanaindicator=yes;
    indicator_enforced=no;
elif test "x$have_ubuntuindicator" = "xyes"; then
    use_ubuntuindicator=yes;
    indicator_enforced=no;
else
    AC_MSG_ERROR([Either Ayatana Indicators or Ubuntu Indicators are required to build mate-indicator-applet.])
fi
```
it doesn't show special messages when one is requested but the other is present, and it selects one over the other if both are present (here, Ayatana's, but it can be swapped easily if wanted -- I have no idea which one should be "best"); but it should cover all cases.

Reply:
@sunweaver: can you please have a look at this...?